### PR TITLE
Revert "Upgrade dependencies to their latest stable release (#15216)"

### PR DESCRIPTION
**Pull request by tudor-da**
This reverts commit 1454497f (due to integration challenges downstream)

But keeps https://github.com/digital-asset/daml/commit/599e3ae4b0d2bb8c46d1b077215ec4c2ed35c08a

changelog_begin
changelog_end

### CODE REVIEW

File: bazel-java-deps.bzl [hunk 1]
```diff
@@ -119,7 +118,7 @@ def install_java_deps():
             "io.opentelemetry:opentelemetry-context:{}".format(opentelemetry_version),
             "io.opentelemetry:opentelemetry-sdk-testing:{}".format(opentelemetry_version),
             "io.opentelemetry:opentelemetry-sdk-trace:{}".format(opentelemetry_version),
-            "io.opentelemetry:opentelemetry-semconv:{}-alpha".format(opentelemetry_version),
+            "io.opentelemetry:opentelemetry-semconv:1.12.0-alpha",
```
Comment (nicu-da) [inline bazel-java-deps.bzl:121]: This one should probs not be reverted

**Review by nicu-da — COMMENTED** [review]
Can you add some more details as to what challenges we encountered? 
We should proceed with keep our dependencies upgraded, so ideally we should try to revert only the libs that we actually need to revert to simplify our upgrade.

File: bazel-java-deps.bzl (hunk 1 above)
Comment (tudor-da) [reply bazel-java-deps.bzl:121]: Wasn't this affected in the original PR?

**Comment by tudor-da** [comment]
> Can you add some more details as to what challenges we encountered?
We should proceed with keep our dependencies upgraded, so ideally we should try to revert only the libs that we actually need to revert to simplify our upgrade.

As motivation, the problem is the integration with Canton, where we have various problems due to updated dependencies. We encountered two problems so far that need a revert of their culprit dependencies:
* Conflicts in sbt dependency resolution for `slf4j-api` which evolved in a backwards incompatible manner wrt to its backends (see https://www.slf4j.org/faq.html#changesInVersion200)
* Cats update of the `traverse_` before (introduced in 2.7.x I believe) which requires major fixes in the Canton codebase.

File: bazel-java-deps.bzl (hunk 1 above)
Comment (nicu-da) [reply bazel-java-deps.bzl:121]: Ah yes, it was initially intended to be merged in another PR but was eventually done in this one. You can ignore

**Comment by nicu-da** [comment]
@tudor-da Any reason not to revert only those dependencies in that case?

**Comment by tudor-da** [comment]
> @tudor-da Any reason not to revert only those dependencies in that case?

We could but not sure if it would be the end of it, so it's probably better to change the strategy. The strategy going forward is to turn this around and start updating the deps from the consumer side.

**Review by rgugliel-da — APPROVED** [review]
Thanks!

**Review by mziolekda — APPROVED** [review]

**Pull request merged**